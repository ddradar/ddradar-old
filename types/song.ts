import {
  hasNumberProperty,
  hasProperty,
  hasStringProperty
} from '../utils/type-assert'

import { isSeries, Series } from './series'

export type Song = {
  /** ^([01689bdiloqDIOPQ]*){32}$ */
  id: string
  name: string
  /** name furigana ^([A-Z0-9 .ぁ-んー]*)$ */
  nameKana: string
  nameIndex: SongIndex
  artist: string
  series: Series
  /** Displayed min BPM (Beet Per Minutes). */
  minBPM: number | null
  /** Displayed max BPM (Beet Per Minutes). */
  maxBPM: number | null
  /** version for diff (yyyymmdd) */
  version: number
}

export type SongIndex =
  | 0
  | 1
  | 2
  | 3
  | 4
  | 5
  | 6
  | 7
  | 8
  | 9
  | 10
  | 11
  | 12
  | 13
  | 14
  | 15
  | 16
  | 17
  | 18
  | 19
  | 20
  | 21
  | 22
  | 23
  | 24
  | 25
  | 26
  | 27
  | 28
  | 29
  | 30
  | 31
  | 32
  | 33
  | 34
  | 35
  | 36

export const SongNameIndex: string[] = [
  'あ',
  'か',
  'さ',
  'た',
  'な',
  'は',
  'ま',
  'や',
  'ら',
  'わ',
  'A',
  'B',
  'C',
  'D',
  'E',
  'F',
  'G',
  'H',
  'I',
  'J',
  'K',
  'L',
  'M',
  'N',
  'O',
  'P',
  'Q',
  'R',
  'S',
  'T',
  'U',
  'V',
  'W',
  'X',
  'Y',
  'Z',
  '数字・記号'
]

export const isSong = (obj: unknown): obj is Song =>
  typeof obj === 'object' &&
  obj !== null &&
  hasStringProperty(obj, 'id', 'name', 'nameKana', 'artist') &&
  /^([A-Z0-9 .ぁ-んー]*)$/.test(obj.nameKana) &&
  hasProperty(obj, 'nameIndex', 'series', 'minBPM', 'maxBPM') &&
  isSongIndex(obj.nameIndex) &&
  isSeries(obj.series) &&
  (typeof obj.minBPM === 'number' || obj.minBPM === null) &&
  (typeof obj.maxBPM === 'number' || obj.maxBPM === null) &&
  hasNumberProperty(obj, 'version')

export const isSongIndex = (obj: unknown): obj is SongIndex =>
  typeof obj === 'number' && Number.isInteger(obj) && obj >= 0 && obj <= 36
