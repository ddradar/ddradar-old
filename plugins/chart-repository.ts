import { Level } from '~/types/level'
import { PlayStyle } from '~/types/play-style'
import { isStepChart, StepChart } from '~/types/step-chart'

export const chartVersion = 20200127

export const fetchSongCharts = async (
  songId: string,
  useMaster: boolean = false
) =>
  (await fetchChartJson(useMaster))
    .filter((c) => c.songId === songId)
    .sort((l, r) =>
      l.playStyle !== r.playStyle
        ? l.playStyle - r.playStyle
        : l.difficulty - r.difficulty
    )

export const fetchChartsByLevel = async (
  playStyle: PlayStyle,
  level: Level,
  useMaster: boolean = false
) =>
  (await fetchChartJson(useMaster))
    .filter((c) => c.level === level && c.playStyle === playStyle)
    .sort((l, r) =>
      l.songName < r.songName
        ? -1
        : l.songName > r.songName
        ? 1
        : l.difficulty - r.difficulty
    )

const masterUrl = 'https://staging.ddradar.app'
const fetchChartJson = async (useMaster: boolean) => {
  const jsonUrl = `${useMaster ? masterUrl : ''}/chart.json`
  const jsonData = await (await fetch(jsonUrl)).json()
  if (Array.isArray(jsonData) && isStepChart(jsonData[0]))
    return jsonData as StepChart[]
  throw new Error('JSON file is not StepChart[]')
}
