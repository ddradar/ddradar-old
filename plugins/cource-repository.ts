import courceJson from '~/static/cource.json'
import { Cource, isCource } from '~/types/cource'
import { Difficulty, getDifficultyName } from '~/types/difficulty'
import { getPlayStyleName, PlayStyle } from '~/types/play-style'

export const version = 20190612

export const fetchCource = (
  courceId: string,
  playStyle: PlayStyle,
  difficulty: Difficulty
): Cource => {
  const cource = courceJson.find((c) => c.id === courceId)
  if (cource === undefined) throw new Error(`Not Found Cource Id: ${courceId}`)
  const order = cource.orders.find(
    (o) => o.playStyle === playStyle && o.difficulty === difficulty
  )
  if (order === undefined)
    throw new Error(
      `Not Found ${getPlayStyleName(playStyle)}/${getDifficultyName(
        difficulty
      )} order`
    )
  return {
    id: cource.id,
    name: cource.name,
    playStyle: order.playStyle,
    difficulty: order.difficulty,
    level: order.level,
    chartOrder: order.chartOrder
  } as Cource
}

export const fetchAllCourceList = () =>
  courceJson
    .flatMap((c) =>
      c.orders.map((o) => ({
        ...o,
        id: c.id,
        name: c.name
      }))
    )
    .filter((c) => isCource(c))
