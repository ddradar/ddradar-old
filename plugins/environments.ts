import { Plugin } from '@nuxt/types'
import { config } from 'dotenv'

config()

export type EnvironmentVariables = {
  NODE_ENV: string
  FIREBASE_API_KEY: string
  FIREBASE_AUTH_DOMAIN: string
  FIREBASE_DATABASE_URL: string
  FIREBASE_PROJECT_ID: string
  FIREBASE_STORAGE_BUCKET: string
  FIREBASE_MESSAGING_SENDER_ID: string
  FIREBASE_APP_ID: string
}

/* eslint-disable no-process-env */
export const environments: EnvironmentVariables = {
  NODE_ENV: process.env.NODE_ENV!,
  FIREBASE_API_KEY: process.env.FIREBASE_API_KEY!,
  FIREBASE_AUTH_DOMAIN: process.env.FIREBASE_AUTH_DOMAIN!,
  FIREBASE_DATABASE_URL: process.env.FIREBASE_DATABASE_URL!,
  FIREBASE_PROJECT_ID: process.env.FIREBASE_PROJECT_ID!,
  FIREBASE_STORAGE_BUCKET: process.env.FIREBASE_STORAGE_BUCKET!,
  FIREBASE_MESSAGING_SENDER_ID: process.env.FIREBASE_MESSAGING_SENDER_ID!,
  FIREBASE_APP_ID: process.env.FIREBASE_APP_ID!
}
/* eslint-enable no-process-env */

/** Validate environments values. */
export const validateEnvironments = ():
  | { valid: true }
  | { valid: false; keys: Extract<keyof EnvironmentVariables, string>[] } => {
  const invalidKeys: string[] = Object.keys(environments).filter((key) => {
    const value: unknown = (environments as any)[key]
    return value === undefined || value === null
  })
  return invalidKeys.length === 0
    ? { valid: true }
    : {
        valid: false,
        keys: invalidKeys as Extract<keyof EnvironmentVariables, string>[]
      }
}

declare module 'vue/types/vue' {
  interface Vue {
    $environments: EnvironmentVariables
  }
}

const environmentsPlugin: Plugin = (_, inject) => {
  inject('environments', environments)
}

export default environmentsPlugin
