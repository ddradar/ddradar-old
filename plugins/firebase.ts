import { Context, Plugin } from '@nuxt/types'
import firebase from 'firebase/app'

const firebasePlugin: Plugin = ({ env }: Pick<Context, 'env'>, _) => {
  if (!firebase.apps.length) {
    firebase.initializeApp({
      apiKey: env.apiKey,
      authDomain: env.authDomain,
      databaseURL: env.databaseURL,
      projectId: env.projectId,
      storageBucket: env.storageBucket,
      messagingSenderId: env.messagingSenderId,
      appId: env.appId
    })
  }
}

export default firebasePlugin
