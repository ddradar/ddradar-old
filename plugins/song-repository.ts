import firebase from '@/plugins/firebase'
import 'firebase/firestore'
import { isSong, Song } from '@/types/song'

const db = firebase.firestore()

export async function fetchSongs(fieldName: string, condition: any) {
  const query = db
    .collection('version/1/songs')
    .where(fieldName, '==', condition)

  const querySorted =
    fieldName === 'nameIndex'
      ? query.orderBy('nameKana')
      : query.orderBy('nameIndex').orderBy('nameKana')
  const snapShot = await querySorted.get()
  return snapShot.docs.map(doc => doc.data()).filter(d => isSong(d)) as Song[]
}

export async function fetchSongById(songId: string) {
  const doc = await db.doc(`version/1/songs/${songId}`).get()
  const unknown = doc.data()
  return isSong(unknown)
    ? unknown
    : Promise.reject(
        new Error(`"version/1/songs/${songId}" is not Song object`)
      )
}
