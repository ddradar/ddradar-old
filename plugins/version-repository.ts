import firebase from '@/plugins/firebase'
import 'firebase/firestore'
import { hasNumberProperty } from '~/test/util'

const db = firebase.firestore()

export async function fetchVersion() {
  const c = await db.doc(`version/1`).get()
  const data = c.data()
  return isVersionDoc(data) ? data : { chartVersion: 0, songVersion: 0 }
}

function isVersionDoc(
  obj: unknown
): obj is { chartVersion: number; songVersion: number } {
  return (
    hasNumberProperty(obj, 'chartVersion') &&
    hasNumberProperty(obj, 'songVersion')
  )
}
