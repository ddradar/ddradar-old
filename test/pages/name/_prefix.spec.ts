import { shallowMount, createLocalVue, Wrapper } from '@vue/test-utils'
import Buefy from 'buefy'
import { mocked } from 'ts-jest/utils'
import NameIndexPage from '@/pages/name/_prefix.vue'
import * as repo from '@/plugins/song-repository'
import { SongNameIndex } from '~/types/song'

jest.mock('@/plugins/song-repository')

const localVue = createLocalVue()
localVue.use(Buefy)

describe('/name/:prefix', () => {
  let wrapper: Wrapper<NameIndexPage>
  let vm: any
  beforeEach(() => {
    wrapper = shallowMount(NameIndexPage, { localVue, stubs: ['song-list'] })
    vm = wrapper.vm
  })
  test('is a Vue instance', () => {
    expect(wrapper.isVueInstance()).toBeTruthy()
  })
  test('selected:null and isLoading:true default', () => {
    expect(vm.selected).toBeNull()
    expect(vm.isLoading).toBe(true)
  })
  describe('asyncData method', () => {
    const hasDataIndex = 0
    const noDataIndex = 1
    const throwErrorIndex = 2
    beforeAll(() => {
      mocked(repo).fetchSongs.mockImplementation((_name, nameIndex) => {
        if (nameIndex === hasDataIndex) {
          return Promise.resolve([
            {
              name: 'PARANOiA',
              nameKana: 'PARANOIA',
              nameIndex: 25,
              artist: '180',
              series: '1st',
              minBPM: 180,
              maxBPM: 180
            }
          ])
        }
        if (nameIndex === noDataIndex) {
          return Promise.resolve([])
        }
        return Promise.reject(new Error('Mock test error'))
      })
    })
    test('selected is null if param is not SongIndex', async () => {
      for (const param of ['', 'hoge', '37', '-1', '1.5', 'NaN']) {
        const data = await vm.$options.asyncData({
          params: { id: param }
        })
        wrapper.setData(data)
        expect(vm.selected).toBeNull()
        expect(vm.isLoading).toBe(false)
      }
    })
    test('selected equals param if param is 0-16', async () => {
      const data = await vm.$options.asyncData({
        params: { prefix: hasDataIndex.toString() }
      })
      wrapper.setData(data)
      expect(vm.selected).toBe(hasDataIndex)
      expect(vm.isLoading).toBe(false)
    })
    test('selected equals param if param is 0-16', async () => {
      const data = await vm.$options.asyncData({
        params: { prefix: noDataIndex.toString() }
      })
      wrapper.setData(data)
      expect(vm.selected).toBe(noDataIndex)
      expect(vm.isLoading).toBe(false)
    })
    test('selected equals param if param is 0-16', async () => {
      const data = await vm.$options.asyncData({
        params: { prefix: throwErrorIndex.toString() }
      })
      wrapper.setData(data)
      expect(vm.selected).toBe(throwErrorIndex)
      expect(vm.isLoading).toBe(false)
    })
  })
  describe('pageTitle getter', () => {
    test('returns "曲名から探す" if series not selected', () => {
      expect(vm.pageTitle).toBe('曲名から探す')
    })
    test('returns SongIndex title if selected', () => {
      for (let i = 0; i < SongNameIndex.length; i++) {
        wrapper.setData({ selected: i })
        expect(vm.pageTitle).toBe(SongNameIndex[i])
      }
    })
  })
  describe('message getter', () => {
    test('returns "曲名を選択してください" if series not selected', () => {
      expect(vm.message).toBe('曲名を選択してください')
    })
    test('returns "Not Found" if songs is empty', () => {
      wrapper.setData({ selected: 0, songs: [] })
      expect(vm.message).toBe('Not Found')
    })
    test('returns "Found n songs" songs is not empty', () => {
      wrapper.setData({
        selected: 0,
        songs: [
          {
            name: 'PARANOiA',
            nameKana: 'PARANOIA',
            nameIndex: 25,
            artist: '180',
            series: '1st',
            minBPM: 180,
            maxBPM: 180
          }
        ]
      })
      expect(vm.message).toBe('Found 1 songs')
      wrapper.setData({
        selected: 0,
        songs: [
          {
            name: 'PARANOiA',
            nameKana: 'PARANOIA',
            nameIndex: 25,
            artist: '180',
            series: '1st',
            minBPM: 180,
            maxBPM: 180
          },
          {
            name: 'PARANOiA',
            nameKana: 'PARANOIA',
            nameIndex: 25,
            artist: '180',
            series: '1st',
            minBPM: 180,
            maxBPM: 180
          }
        ]
      })
      expect(vm.message).toBe('Found 2 songs')
    })
  })
})
