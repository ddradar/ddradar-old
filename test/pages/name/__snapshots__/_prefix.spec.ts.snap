// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`/name/:prefix renders correctly 1`] = `
<section
  class="section"
>
  <h1
    class="title"
  >
    数字・記号
  </h1>
   
  <div
    class="buttons"
  >
    <a
      class="button is-info"
      type="button"
    >
      <!---->
       
      <span>
        
      あ
    
      </span>
       
      <!---->
    </a>
    <a
      class="button is-info"
      type="button"
    >
      <!---->
       
      <span>
        
      か
    
      </span>
       
      <!---->
    </a>
    <a
      class="button is-info"
      type="button"
    >
      <!---->
       
      <span>
        
      さ
    
      </span>
       
      <!---->
    </a>
    <a
      class="button is-info"
      type="button"
    >
      <!---->
       
      <span>
        
      た
    
      </span>
       
      <!---->
    </a>
    <a
      class="button is-info"
      type="button"
    >
      <!---->
       
      <span>
        
      な
    
      </span>
       
      <!---->
    </a>
    <a
      class="button is-info"
      type="button"
    >
      <!---->
       
      <span>
        
      は
    
      </span>
       
      <!---->
    </a>
    <a
      class="button is-info"
      type="button"
    >
      <!---->
       
      <span>
        
      ま
    
      </span>
       
      <!---->
    </a>
    <a
      class="button is-info"
      type="button"
    >
      <!---->
       
      <span>
        
      や
    
      </span>
       
      <!---->
    </a>
    <a
      class="button is-info"
      type="button"
    >
      <!---->
       
      <span>
        
      ら
    
      </span>
       
      <!---->
    </a>
    <a
      class="button is-info"
      type="button"
    >
      <!---->
       
      <span>
        
      わ
    
      </span>
       
      <!---->
    </a>
    <a
      class="button is-info"
      type="button"
    >
      <!---->
       
      <span>
        
      A
    
      </span>
       
      <!---->
    </a>
    <a
      class="button is-info"
      type="button"
    >
      <!---->
       
      <span>
        
      B
    
      </span>
       
      <!---->
    </a>
    <a
      class="button is-info"
      type="button"
    >
      <!---->
       
      <span>
        
      C
    
      </span>
       
      <!---->
    </a>
    <a
      class="button is-info"
      type="button"
    >
      <!---->
       
      <span>
        
      D
    
      </span>
       
      <!---->
    </a>
    <a
      class="button is-info"
      type="button"
    >
      <!---->
       
      <span>
        
      E
    
      </span>
       
      <!---->
    </a>
    <a
      class="button is-info"
      type="button"
    >
      <!---->
       
      <span>
        
      F
    
      </span>
       
      <!---->
    </a>
    <a
      class="button is-info"
      type="button"
    >
      <!---->
       
      <span>
        
      G
    
      </span>
       
      <!---->
    </a>
    <a
      class="button is-info"
      type="button"
    >
      <!---->
       
      <span>
        
      H
    
      </span>
       
      <!---->
    </a>
    <a
      class="button is-info"
      type="button"
    >
      <!---->
       
      <span>
        
      I
    
      </span>
       
      <!---->
    </a>
    <a
      class="button is-info"
      type="button"
    >
      <!---->
       
      <span>
        
      J
    
      </span>
       
      <!---->
    </a>
    <a
      class="button is-info"
      type="button"
    >
      <!---->
       
      <span>
        
      K
    
      </span>
       
      <!---->
    </a>
    <a
      class="button is-info"
      type="button"
    >
      <!---->
       
      <span>
        
      L
    
      </span>
       
      <!---->
    </a>
    <a
      class="button is-info"
      type="button"
    >
      <!---->
       
      <span>
        
      M
    
      </span>
       
      <!---->
    </a>
    <a
      class="button is-info"
      type="button"
    >
      <!---->
       
      <span>
        
      N
    
      </span>
       
      <!---->
    </a>
    <a
      class="button is-info"
      type="button"
    >
      <!---->
       
      <span>
        
      O
    
      </span>
       
      <!---->
    </a>
    <a
      class="button is-info"
      type="button"
    >
      <!---->
       
      <span>
        
      P
    
      </span>
       
      <!---->
    </a>
    <a
      class="button is-info"
      type="button"
    >
      <!---->
       
      <span>
        
      Q
    
      </span>
       
      <!---->
    </a>
    <a
      class="button is-info"
      type="button"
    >
      <!---->
       
      <span>
        
      R
    
      </span>
       
      <!---->
    </a>
    <a
      class="button is-info"
      type="button"
    >
      <!---->
       
      <span>
        
      S
    
      </span>
       
      <!---->
    </a>
    <a
      class="button is-info"
      type="button"
    >
      <!---->
       
      <span>
        
      T
    
      </span>
       
      <!---->
    </a>
    <a
      class="button is-info"
      type="button"
    >
      <!---->
       
      <span>
        
      U
    
      </span>
       
      <!---->
    </a>
    <a
      class="button is-info"
      type="button"
    >
      <!---->
       
      <span>
        
      V
    
      </span>
       
      <!---->
    </a>
    <a
      class="button is-info"
      type="button"
    >
      <!---->
       
      <span>
        
      W
    
      </span>
       
      <!---->
    </a>
    <a
      class="button is-info"
      type="button"
    >
      <!---->
       
      <span>
        
      X
    
      </span>
       
      <!---->
    </a>
    <a
      class="button is-info"
      type="button"
    >
      <!---->
       
      <span>
        
      Y
    
      </span>
       
      <!---->
    </a>
    <a
      class="button is-info"
      type="button"
    >
      <!---->
       
      <span>
        
      Z
    
      </span>
       
      <!---->
    </a>
    <button
      class="button is-info is-outlined"
      disabled="disabled"
      to="/name/36"
      type="button"
    >
      <!---->
       
      <span>
        
      数字・記号
    
      </span>
       
      <!---->
    </button>
  </div>
   
  <p
    class="subtitle"
  >
    Found 1 songs
  </p>
   
  <songlist-stub
    loading="true"
    songs="[object Object]"
  />
</section>
`;

exports[`/name/:prefix renders not found 1`] = `
<section
  class="section"
>
  <h1
    class="title"
  >
    数字・記号
  </h1>
   
  <div
    class="buttons"
  >
    <a
      class="button is-info"
      type="button"
    >
      <!---->
       
      <span>
        
      あ
    
      </span>
       
      <!---->
    </a>
    <a
      class="button is-info"
      type="button"
    >
      <!---->
       
      <span>
        
      か
    
      </span>
       
      <!---->
    </a>
    <a
      class="button is-info"
      type="button"
    >
      <!---->
       
      <span>
        
      さ
    
      </span>
       
      <!---->
    </a>
    <a
      class="button is-info"
      type="button"
    >
      <!---->
       
      <span>
        
      た
    
      </span>
       
      <!---->
    </a>
    <a
      class="button is-info"
      type="button"
    >
      <!---->
       
      <span>
        
      な
    
      </span>
       
      <!---->
    </a>
    <a
      class="button is-info"
      type="button"
    >
      <!---->
       
      <span>
        
      は
    
      </span>
       
      <!---->
    </a>
    <a
      class="button is-info"
      type="button"
    >
      <!---->
       
      <span>
        
      ま
    
      </span>
       
      <!---->
    </a>
    <a
      class="button is-info"
      type="button"
    >
      <!---->
       
      <span>
        
      や
    
      </span>
       
      <!---->
    </a>
    <a
      class="button is-info"
      type="button"
    >
      <!---->
       
      <span>
        
      ら
    
      </span>
       
      <!---->
    </a>
    <a
      class="button is-info"
      type="button"
    >
      <!---->
       
      <span>
        
      わ
    
      </span>
       
      <!---->
    </a>
    <a
      class="button is-info"
      type="button"
    >
      <!---->
       
      <span>
        
      A
    
      </span>
       
      <!---->
    </a>
    <a
      class="button is-info"
      type="button"
    >
      <!---->
       
      <span>
        
      B
    
      </span>
       
      <!---->
    </a>
    <a
      class="button is-info"
      type="button"
    >
      <!---->
       
      <span>
        
      C
    
      </span>
       
      <!---->
    </a>
    <a
      class="button is-info"
      type="button"
    >
      <!---->
       
      <span>
        
      D
    
      </span>
       
      <!---->
    </a>
    <a
      class="button is-info"
      type="button"
    >
      <!---->
       
      <span>
        
      E
    
      </span>
       
      <!---->
    </a>
    <a
      class="button is-info"
      type="button"
    >
      <!---->
       
      <span>
        
      F
    
      </span>
       
      <!---->
    </a>
    <a
      class="button is-info"
      type="button"
    >
      <!---->
       
      <span>
        
      G
    
      </span>
       
      <!---->
    </a>
    <a
      class="button is-info"
      type="button"
    >
      <!---->
       
      <span>
        
      H
    
      </span>
       
      <!---->
    </a>
    <a
      class="button is-info"
      type="button"
    >
      <!---->
       
      <span>
        
      I
    
      </span>
       
      <!---->
    </a>
    <a
      class="button is-info"
      type="button"
    >
      <!---->
       
      <span>
        
      J
    
      </span>
       
      <!---->
    </a>
    <a
      class="button is-info"
      type="button"
    >
      <!---->
       
      <span>
        
      K
    
      </span>
       
      <!---->
    </a>
    <a
      class="button is-info"
      type="button"
    >
      <!---->
       
      <span>
        
      L
    
      </span>
       
      <!---->
    </a>
    <a
      class="button is-info"
      type="button"
    >
      <!---->
       
      <span>
        
      M
    
      </span>
       
      <!---->
    </a>
    <a
      class="button is-info"
      type="button"
    >
      <!---->
       
      <span>
        
      N
    
      </span>
       
      <!---->
    </a>
    <a
      class="button is-info"
      type="button"
    >
      <!---->
       
      <span>
        
      O
    
      </span>
       
      <!---->
    </a>
    <a
      class="button is-info"
      type="button"
    >
      <!---->
       
      <span>
        
      P
    
      </span>
       
      <!---->
    </a>
    <a
      class="button is-info"
      type="button"
    >
      <!---->
       
      <span>
        
      Q
    
      </span>
       
      <!---->
    </a>
    <a
      class="button is-info"
      type="button"
    >
      <!---->
       
      <span>
        
      R
    
      </span>
       
      <!---->
    </a>
    <a
      class="button is-info"
      type="button"
    >
      <!---->
       
      <span>
        
      S
    
      </span>
       
      <!---->
    </a>
    <a
      class="button is-info"
      type="button"
    >
      <!---->
       
      <span>
        
      T
    
      </span>
       
      <!---->
    </a>
    <a
      class="button is-info"
      type="button"
    >
      <!---->
       
      <span>
        
      U
    
      </span>
       
      <!---->
    </a>
    <a
      class="button is-info"
      type="button"
    >
      <!---->
       
      <span>
        
      V
    
      </span>
       
      <!---->
    </a>
    <a
      class="button is-info"
      type="button"
    >
      <!---->
       
      <span>
        
      W
    
      </span>
       
      <!---->
    </a>
    <a
      class="button is-info"
      type="button"
    >
      <!---->
       
      <span>
        
      X
    
      </span>
       
      <!---->
    </a>
    <a
      class="button is-info"
      type="button"
    >
      <!---->
       
      <span>
        
      Y
    
      </span>
       
      <!---->
    </a>
    <a
      class="button is-info"
      type="button"
    >
      <!---->
       
      <span>
        
      Z
    
      </span>
       
      <!---->
    </a>
    <button
      class="button is-info is-outlined"
      disabled="disabled"
      to="/name/36"
      type="button"
    >
      <!---->
       
      <span>
        
      数字・記号
    
      </span>
       
      <!---->
    </button>
  </div>
   
  <p
    class="subtitle"
  >
    Not Found
  </p>
   
  <songlist-stub
    loading="true"
    songs=""
  />
</section>
`;

exports[`/name/:prefix renders select SongIndex 1`] = `
<section
  class="section"
>
  <h1
    class="title"
  >
    曲名から探す
  </h1>
   
  <div
    class="buttons"
  >
    <a
      class="button is-info"
      type="button"
    >
      <!---->
       
      <span>
        
      あ
    
      </span>
       
      <!---->
    </a>
    <a
      class="button is-info"
      type="button"
    >
      <!---->
       
      <span>
        
      か
    
      </span>
       
      <!---->
    </a>
    <a
      class="button is-info"
      type="button"
    >
      <!---->
       
      <span>
        
      さ
    
      </span>
       
      <!---->
    </a>
    <a
      class="button is-info"
      type="button"
    >
      <!---->
       
      <span>
        
      た
    
      </span>
       
      <!---->
    </a>
    <a
      class="button is-info"
      type="button"
    >
      <!---->
       
      <span>
        
      な
    
      </span>
       
      <!---->
    </a>
    <a
      class="button is-info"
      type="button"
    >
      <!---->
       
      <span>
        
      は
    
      </span>
       
      <!---->
    </a>
    <a
      class="button is-info"
      type="button"
    >
      <!---->
       
      <span>
        
      ま
    
      </span>
       
      <!---->
    </a>
    <a
      class="button is-info"
      type="button"
    >
      <!---->
       
      <span>
        
      や
    
      </span>
       
      <!---->
    </a>
    <a
      class="button is-info"
      type="button"
    >
      <!---->
       
      <span>
        
      ら
    
      </span>
       
      <!---->
    </a>
    <a
      class="button is-info"
      type="button"
    >
      <!---->
       
      <span>
        
      わ
    
      </span>
       
      <!---->
    </a>
    <a
      class="button is-info"
      type="button"
    >
      <!---->
       
      <span>
        
      A
    
      </span>
       
      <!---->
    </a>
    <a
      class="button is-info"
      type="button"
    >
      <!---->
       
      <span>
        
      B
    
      </span>
       
      <!---->
    </a>
    <a
      class="button is-info"
      type="button"
    >
      <!---->
       
      <span>
        
      C
    
      </span>
       
      <!---->
    </a>
    <a
      class="button is-info"
      type="button"
    >
      <!---->
       
      <span>
        
      D
    
      </span>
       
      <!---->
    </a>
    <a
      class="button is-info"
      type="button"
    >
      <!---->
       
      <span>
        
      E
    
      </span>
       
      <!---->
    </a>
    <a
      class="button is-info"
      type="button"
    >
      <!---->
       
      <span>
        
      F
    
      </span>
       
      <!---->
    </a>
    <a
      class="button is-info"
      type="button"
    >
      <!---->
       
      <span>
        
      G
    
      </span>
       
      <!---->
    </a>
    <a
      class="button is-info"
      type="button"
    >
      <!---->
       
      <span>
        
      H
    
      </span>
       
      <!---->
    </a>
    <a
      class="button is-info"
      type="button"
    >
      <!---->
       
      <span>
        
      I
    
      </span>
       
      <!---->
    </a>
    <a
      class="button is-info"
      type="button"
    >
      <!---->
       
      <span>
        
      J
    
      </span>
       
      <!---->
    </a>
    <a
      class="button is-info"
      type="button"
    >
      <!---->
       
      <span>
        
      K
    
      </span>
       
      <!---->
    </a>
    <a
      class="button is-info"
      type="button"
    >
      <!---->
       
      <span>
        
      L
    
      </span>
       
      <!---->
    </a>
    <a
      class="button is-info"
      type="button"
    >
      <!---->
       
      <span>
        
      M
    
      </span>
       
      <!---->
    </a>
    <a
      class="button is-info"
      type="button"
    >
      <!---->
       
      <span>
        
      N
    
      </span>
       
      <!---->
    </a>
    <a
      class="button is-info"
      type="button"
    >
      <!---->
       
      <span>
        
      O
    
      </span>
       
      <!---->
    </a>
    <a
      class="button is-info"
      type="button"
    >
      <!---->
       
      <span>
        
      P
    
      </span>
       
      <!---->
    </a>
    <a
      class="button is-info"
      type="button"
    >
      <!---->
       
      <span>
        
      Q
    
      </span>
       
      <!---->
    </a>
    <a
      class="button is-info"
      type="button"
    >
      <!---->
       
      <span>
        
      R
    
      </span>
       
      <!---->
    </a>
    <a
      class="button is-info"
      type="button"
    >
      <!---->
       
      <span>
        
      S
    
      </span>
       
      <!---->
    </a>
    <a
      class="button is-info"
      type="button"
    >
      <!---->
       
      <span>
        
      T
    
      </span>
       
      <!---->
    </a>
    <a
      class="button is-info"
      type="button"
    >
      <!---->
       
      <span>
        
      U
    
      </span>
       
      <!---->
    </a>
    <a
      class="button is-info"
      type="button"
    >
      <!---->
       
      <span>
        
      V
    
      </span>
       
      <!---->
    </a>
    <a
      class="button is-info"
      type="button"
    >
      <!---->
       
      <span>
        
      W
    
      </span>
       
      <!---->
    </a>
    <a
      class="button is-info"
      type="button"
    >
      <!---->
       
      <span>
        
      X
    
      </span>
       
      <!---->
    </a>
    <a
      class="button is-info"
      type="button"
    >
      <!---->
       
      <span>
        
      Y
    
      </span>
       
      <!---->
    </a>
    <a
      class="button is-info"
      type="button"
    >
      <!---->
       
      <span>
        
      Z
    
      </span>
       
      <!---->
    </a>
    <a
      class="button is-info"
      type="button"
    >
      <!---->
       
      <span>
        
      数字・記号
    
      </span>
       
      <!---->
    </a>
  </div>
   
  <p
    class="subtitle"
  >
    曲名を選択してください
  </p>
   
  <!---->
</section>
`;
